File: backend/KhataFlow.WebAPI.Solution/KhataFlow.Core/DTO/Response/CustomerLedgerViewResponse.cs
namespace KhataFlow.Core.DTO.Response;

public class CustomerLedgerViewResponse
{
    public string CustomerName { get; set; } = string.Empty;
    public string CustomerNameUr { get; set; } = string.Empty;
    public string BusinessName { get; set; } = string.Empty;
    public string BusinessNameUr { get; set; } = string.Empty;
    public decimal CurrentBalance { get; set; } 
    public string Currency { get; set; } = "PKR";
    public List<LedgerHistoryItemResponse> History { get; set; } = new();
}

File: frontend/src/app/core/models/customer-ledger-view-response.ts
export interface CustomerLedgerViewLineItem {
  productName: string;
  productNameUr: string;
  quantity: number;
  unitPrice: number;
  lineTotal: number;
}

export interface CustomerLedgerViewEntry {
  type: 'Udhar' | 'Cash';
  amount: number;
  date: string;
  description: string;
  runningBalance: number;
  items: CustomerLedgerViewLineItem[] | null;
}

export interface CustomerLedgerViewResponse {
  customerName: string;
  customerNameUr: string | null;
  businessName: string;
  businessNameUr: string | null;
  currentBalance: number;
  currency: string;
  history: CustomerLedgerViewEntry[];
}